components/api/src/main/java/com/hotels/styx/api/extension/service/BackendService.java

```diff
@@ -382,7 +390,7 @@ public Builder rewrites(RewriteConfig... rewriteConfigs) {
          * @return this builder
          */
         public Builder rewrites(List<RewriteConfig> rewriteConfigs) {
-            this.rewrites = ImmutableList.copyOf(rewriteConfigs);
+            this.rewrites = unmodifiableList(rewriteConfigs.stream().map(Objects::requireNonNull).collect(toList()));
```

Comment:
IIRC we use a `copyOf` method almost everywhere. Do you think we can implement our own version of it? (Perhaps not necessarily for the API module, but for remaining styx).